File: levelup-backend/src/main/java/com/levelup/backend/service/EventService.java
package com.levelup.backend.service;

import com.levelup.backend.entity.Event;
import com.levelup.backend.repository.EventRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class EventService {

    private final EventRepository eventRepository;

    public List<Event> getAllEvents() {
        return eventRepository.findAll();
    }

    public Event getEventById(Long id) {
        return eventRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Evento no encontrado"));
    }

    public Event createEvent(Event event) {
        return eventRepository.save(event);
    }

    public Event updateEvent(Long id, Event eventDetails) {
        Event event = getEventById(id);
        event.setTitulo(eventDetails.getTitulo());
        event.setDescripcion(eventDetails.getDescripcion());
        event.setFecha(eventDetails.getFecha());
        event.setUbicacion(eventDetails.getUbicacion());
        event.setImagen(eventDetails.getImagen());
        // Update other fields
        return eventRepository.save(event);
    }

    public void deleteEvent(Long id) {
        eventRepository.deleteById(id);
    }
}

File: levelup-admin-panel/src/services/EventService.ts
import api from './api';
import { Event } from '../types/Event';

const EventService = {
    getAllEvents: async (): Promise<Event[]> => {
        const response = await api.get('/events');
        return response.data;
    },

    getEventById: async (id: string): Promise<Event> => {
        const response = await api.get(`/events/${id}`);
        return response.data;
    },

    createEvent: async (event: Event): Promise<Event> => {
        const response = await api.post('/events', event);
        return response.data;
    },

    updateEvent: async (id: string, event: Event): Promise<Event> => {
        const response = await api.put(`/events/${id}`, event);
        return response.data;
    },

    deleteEvent: async (id: string): Promise<void> => {
        await api.delete(`/events/${id}`);
    },
};

export default EventService;
